### PR TITLE
Generalize dedicated server auth settings and add a glyph to the server list.

### LINKED ISSUE #15455: Auth-required games should show a glyph in the server list
Requires master server changes.

### PR DESCRIPTION
**Pull request by pchote**
Fixes #15455.

~~Depends on #15498 and https://github.com/OpenRA/OpenRAMasterServer/pull/57~~

In addition to the glyph, this replaces/expands the `RequireAuthIDs` list with separate settings for ID white and black lists, and a default toggle for requiring auth without a specific whitelist.

### CODE REVIEW

**Comment by pchote** [comment]
Rebased and all dependencies have been merged/deployed.

File: OpenRA.Game/Network/GameServer.cs [hunk 1]
```diff
@@ -33,6 +34,7 @@ public class GameClient
 		public GameClient(Session.Client c)
 		{
 			Name = c.Name;
+			Fingerprint = c.Fingerprint;
```
Comment (reaperrr) [inline OpenRA.Game/Network/GameServer.cs]: Just curious, where is this used? I might have just missed it, but it doesn't look like any of this PR's other changes use this.

Comment (pchote) [reply OpenRA.Game/Network/GameServer.cs]: It's not used yet, but at some point in the future we may want to expose registered accounts in the game or web server browsers.

**Review by reaperrr — APPROVED** [review]
Looks good to me, and glyph shows up correctly in all mods

**Review by GraionDilach — APPROVED** [review]
LGTM.

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/ac8e7ba28042b3e68154e0ecb80e3a8117ad1f6c...153a3582c8a4f1823e4a19436a940f98b226e2b3

and picked to prep.

**Pull request merged**